adversarial/advbox/models/paddle.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+
+import numpy as np
+import paddle.v2 as paddle
+import paddle.v2.fluid as fluid
+from paddle.v2.fluid.framework import program_guard
+
+from .base import Model
+
+
+class PaddleModel(Model):
+    """
+    Create a PaddleModel instance.
+    When you need to generate a adversarial sample, you should construct an instance of PaddleModel.
+
+    Args:
+        program(paddle.v2.fluid.framework.Program): The program of the model which generate the adversarial sample.
+        input_name(string): The name of the input.
+        logits_name(string): The name of the logits.
```

Comment:
adversarial attack has two types, untargeted and targeted.
* The goal of the non-targeted attack is to slightly modify source image in a way that image will be classified incorrectly
* The goal of the targeted attack is to slightly modify source image in a way that image will be classified as specified target class

In untargeted attack which we has implemented, the fake label logits make no effect.
In targeted attack, the logits_name means the fake label logits to cheat NN.We will implement the targeted attack later.

Reply:
In my understanding, even in the non-targeted attack, the fake label logits are required to generate the loss. Otherwise, we can neither get the backward gradient nor modify input images further. However, what the fake label logits are is not important. They can be all zeros, random vectors or even different each batch.

Do I understand it correctly?

Reply:
yes